builder/main.py

```diff
@@ -159,11 +305,27 @@ def generate_uf2(target, source, env):
         UPLOADCMD='"$UPLOADER" $UPLOADERFLAGS $SOURCES'
     )
 
+    if "uploadfs" in COMMAND_LINE_TARGETS:
+        env.Replace(
+            UPLOADER=join(platform.get_package_dir("tool-rp2040tools") or "", "picotool"),
+            UPLOADERFLAGS=[
+                "load",
+                "--verify"
+            ],
+            UPLOADCMD='"$UPLOADER" $UPLOADERFLAGS $SOURCES --offset ${hex(FS_START)}',
+        )
+
     upload_actions = [
         env.VerboseAction(BeforeUpload, "Looking for upload port..."),
         env.VerboseAction("$UPLOADCMD", "Uploading $SOURCE"),
     ]
 
+    # picotool seems to need just a tiny bit of delay, but rp2040 load not..
+    if "uploadfs" in COMMAND_LINE_TARGETS:
+        upload_actions.insert(1, env.VerboseAction(DelayBeforeUpload, "Delaying a tiny bit..."))
```

Comment:
Does it make to even further simplify the upload actions by replacing a standalone delay function with a simple lambda like you did above with `_update_max_upload_size`? Something like:
```
env.VerboseAction(
    lambda source, target, env: time.sleep(0.5), "Delaying a tiny bit..."
),
```

Reply:
Function inlined in latest change.